### ISSUE TITLE
About the training and testing setting.

### ISSUE THREAD

**Issue by haoweiz23** [sconlyshootery/FeatDepth#73]
There are many difference setting between FeatDepth & Monodepth2. like Gpus, bs, training epochs, lr. What is your basis for modifying these settings from Monodepth2 ?

Have your ever made comparison experiments?

Besides, I wonder if the reported performance choice the best weights from different epochs in training phase according to the loss on the test set? As I konw, Monodepth2 used the last epoch for their reported results.

**Comment by sconlyshootery** [comment]
You are ridiculous. We have different architecture, why we should have the same settings as monodepth2?
Why we have to choose the best weight according to the test set performance?

**Comment by haoweiz23** [comment]
@sconlyshootery  I just wonder have you ever try their training setting and whether performance will be different. Cause you note in [https://github.com/sconlyshootery/FeatDepth/issues/48](url) that you choose the best weight according to the loss on the test set. So I wonder if you do the same on DepthNet and PoseNet.

**Comment by Zhu-haow** [comment]
> I just wonder have you ever try their training setting and whether performance will be different. Cause you note in [https://github.com/sconlyshootery/FeatDepth/issues/48](url) that you choose the best weight according to the loss on the test set. So I wonder if you do the same on DepthNet and PoseNet.

**Comment by sconlyshootery** [comment]
I can not find the issue. 
I tried other settings, but they are not exactly the same as theirs, and the performance wouldn't change largely when settings change.

**Comment by haoweiz23** [comment]
> I can not find the issue. I tried other settings, but they are not exactly the same as theirs, and the performance wouldn't change largely when settings change.

Sorry, maybe link is invalid.  Thanks for the reply. The title of cited issue is "How to choose featurenet when training separately? ".  [Link](https://github.com/sconlyshootery/FeatDepth/issues/48#issuecomment-817262053)

It metioned that "When traning a featurenet, one could store its weight at different epoches. And how to choose the best weight to restore at following traning of posenet and depthnet? " and you replied "According to the loss on the test set."

So maybe just like Featurenet, the saved weights of DepthNet and PoseNet are also the best epoch with lowest loss on the test set rather than last epoch in training?
